railties/test/rails_info_controller_test.rb

```diff
@@ -56,30 +79,134 @@ def setup
+    assert exact_results.include? "/test/nested_route(.:format)"
+    assert exact_results.include? "/rails/info/properties(.:format)"
+    assert exact_results.include? "/rails/info/routes(.:format)"
+  end
+
+  test "info controller search returns exact matches for route Controller#Action(s)" do
+    get :routes, params: { query: "rails/info#propertie" }
+    assert exact_results.size == 0, "should not match incomplete route Controller#Action(s)"
 
-    get :routes, params: { path: "rails/info/routes" }
-    assert exact_count.call == 1, "should match complete routes"
+    get :routes, params: { query: "rails/info#properties" }
+    assert exact_results.size == 3, "should match complete route Controller#Action(s)"
+    assert exact_results.include? "/rails/info/properties(.:format)"
+    assert exact_results.include? "/rails/:test/properties(.:format)"
+    assert exact_results.include? "/rails/:test/named_properties(.:format)"
+  end
 
-    get :routes, params: { path: "rails/info/routes.html" }
-    assert exact_count.call == 1, "should match complete routes with optional parts"
+  test "info controller returns fuzzy matches for route names" do
+    get :routes, params: { query: "" }
+    assert exact_results.size == 0, "should not match unnamed routes"
+
+    get :routes, params: { query: "rails_info" }
+    assert fuzzy_results.size == 3, "should match incomplete route names"
+    assert fuzzy_results.include? "/rails/info/properties(.:format)"
+    assert fuzzy_results.include? "/rails/info/routes(.:format)"
+    assert fuzzy_results.include? "/rails/:test/named_properties(.:format)"
+
+    get :routes, params: { query: "/rails/info/routes" }
+    assert fuzzy_results.size == 1, "should match complete route names"
+    assert fuzzy_results.include? "/rails/info/routes(.:format)"
+
+    get :routes, params: { query: "named_rails_info_properties_path" }
+    assert fuzzy_results.size == 1, "should match complete route paths"
+    assert fuzzy_results.include? "/rails/:test/named_properties(.:format)"
+
+    get :routes, params: { query: "named_rails_info_properties_url" }
+    assert fuzzy_results.size == 1, "should match complete route urls"
+    assert fuzzy_results.include? "/rails/:test/named_properties(.:format)"
+
+    # Implementing these tests would make the search slightly more consistent if anyone wants to improve the regex implementation
+    # get :routes, params: { query: "rails_info_properties_pat" }
+    # assert fuzzy_results.size == 2, "should match incomplete route paths"
+    # assert fuzzy_results.include? "/rails/info/properties(.:format)"
+    # assert fuzzy_results.include? "/rails/:test/named_properties(.:format)"
+
+    # get :routes, params: { query: "rails_info_properties_ur" }
+    # assert fuzzy_results.size == 2, "should match incomplete route urls"
+    # assert fuzzy_results.include? "/rails/info/properties(.:format)"
+    # assert fuzzy_results.include? "/rails/:test/named_properties(.:format)"
```

Comment:
```suggestion
```

Reply:
Yep, makes sense - just wanted to document this as part of the pull request but will remove it.